programs/drift/src/controller/pnl.rs

```diff
@@ -260,6 +262,37 @@ pub fn settle_expired_position(
         }
     };
 
+    if user.perp_positions[position_index].is_lp() {
+        let margin_calc = calculate_margin_requirement_and_total_collateral_and_liability_info(
+            user,
+            perp_market_map,
+            spot_market_map,
+            oracle_map,
+            MarginContext::standard(MarginRequirementType::Initial),
+        )?;
+
+        attempt_burn_user_lp_shares_for_risk_reduction(
```

Comment:
i thought you said if it doesn't meet margin requirement, it calls this and then returns?